src/reason_pprint_ast.ml

```diff
@@ -2446,6 +2446,62 @@ class printer  ()= object(self:'self)
     in
     (SourceMap (ptype_loc, everything))
 
+  method formatOneTypeExt prepend name assignToken te =
+    let equalInitiatedSegments =
+      let segments = List.map self#type_extension_binding_segments te.ptyext_constructors in
+      [makeList ~break:Always_rec ~postSpace:true ~inline:(true, true) segments] in
+    let formattedTypeParams = List.map self#type_param te.ptyext_params in
+    let binding = makeList ~postSpace:true (prepend::name::[]) in
+    let labelWithParams = match formattedTypeParams with
+        [] -> binding
+      | phd::ptl -> label ~space:true binding (makeList ~postSpace:true ~break:IfNeed ~inline:(true, true) (phd::ptl)) in
+    let everything =
+      let nameParamsEquals = makeList ~postSpace:true [labelWithParams; assignToken] in
+      formatAttachmentApplication
+             typeApplicationFinalWrapping
+             (Some (true, nameParamsEquals))
+             (equalInitiatedSegments, None)
+    in
+    SourceMap (te.ptyext_path.loc, everything)
+
+  method type_extension_binding_segments {pext_kind; pext_loc; pext_attributes; pext_name} =
+    let normalize lst = match lst with
+        | [] -> raise (NotPossible "should not be called")
+        | [hd] -> hd
+        | _::_ -> makeList ~break:Never ~postSpace:true lst (* ~inline:(true, true) *)
+      in
+      let add_bar name args =
+        let lbl = label ~space:true name args in
+        makeList ~postSpace:true [atom "|"; lbl]
+      in
+    let sourceMappedName = SourceMap (pext_name.loc, atom pext_name.txt) in
+    let nameOf = makeList ~postSpace:true [sourceMappedName] in
+    let barName = makeList ~postSpace:true [atom "|"; sourceMappedName] in
+    let resolved = match pext_kind with
+      | Pext_decl (ctor_args, gadt) ->
+        let args = (List.map self#non_arrowed_simple_core_type ctor_args) in
+        let gadtRes = match gadt with
+        | None -> None
+        | Some x -> Some (
+            makeList [
+              formatJustTheTypeConstraint (self#core_type x)
+            ]
+          ) in
+        (args, gadtRes)
+      (* type bar += Foo = Attr.Foo *)
+      | Pext_rebind rebind ->
+        let r = self#longident_loc rebind in
+        let prepend = (atom "=") in
+        ([makeList ~postSpace:true [prepend; r]], None)
+    in
+    let everything = match resolved with
+      | ([], None) -> barName
+      | ([], Some res) -> add_bar sourceMappedName res
+      | (_::_ as args, None) -> add_bar nameOf (normalize args)
+      | (_::_ as args, Some res) -> add_bar nameOf (normalize (args@[res]))
```

Comment:
The first element of the tuple represents constructor arguments, the second an optional formatted gadt.

```ocaml
(* Case 1: No constructor arguments, neither a gadt *)
type attr = ..;
type attr += | Str

(* Case 2: No constructor arguments, is a gadt  *)
type attr = ..;
type attr += | Str :attr

(* Case 3: Has Constructor args, not a gadt  *)
type attr  = ..;
type attr += | Str string;
type attr += | Point int int;

(* Case 4: Has Constructor args & is a gadt *)
type attr  = ..;
type attr += | Point int int :attr;
```